cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -165,6 +165,8 @@ protected void run() {
           } catch (IOException e) {
             OUTAGE_LOG.warn("Failed to fetch or process messages from {}:{}. Will be retried in next iteration.",
                             topic, partition, e);
+            // to avoid busy wait during kafka server outage
+            TimeUnit.SECONDS.sleep(1);
```

Comment:
The sleep shouldn't be unconditional, as there could be messages in the queue. I think the sleep is already handled below.